src/servers/grpc_server.cc

```diff
@@ -249,45 +250,71 @@ GRPCServer::~GRPCServer()
 
 Status
 GRPCServer::Create(
-    InferenceServer* server, uint16_t port,
-    std::unique_ptr<GRPCServer>* grpc_server)
+    InferenceServer* server,
+    const std::map<int32_t, std::vector<std::string>>& port_map,
+    std::vector<std::unique_ptr<GRPCServer>>* grpc_servers)
 {
-  // DLIS-162 - provide global defaults and cli overridable options
+  // size_t i = 0;
+  if (port_map.empty()) {
+    return Status(
+        RequestStatusCode::INVALID_ARG,
+        "GRPC is enabled but none of the service endpoints have a valid port "
+        "assignment");
+  }
   g_Resources = std::make_shared<AsyncResources>(
-      server,  // InferenceServer*,
-      1,       // infer threads
-      1        // mgmt threads
-  );
-
-  LOG_INFO << "Building nvrpc server";
-  const std::string addr = "0.0.0.0:" + std::to_string(port);
-  grpc_server->reset(new GRPCServer(addr));
-
-  (*grpc_server)->GetBuilder().SetMaxMessageSize(MAX_GRPC_MESSAGE_SIZE);
-
-  LOG_INFO << "Register TensorRT GRPCService";
-  auto inferenceService = (*grpc_server)->RegisterAsyncService<GRPCService>();
-
-  LOG_INFO << "Register Infer RPC";
-  (*grpc_server)->rpcInfer_ = inferenceService->RegisterRPC<InferContext>(
-      &GRPCService::AsyncService::RequestInfer);
-
-  LOG_INFO << "Register StreamInfer RPC";
-  (*grpc_server)->rpcStreamInfer_ =
-      inferenceService->RegisterRPC<StreamInferContext>(
-          &GRPCService::AsyncService::RequestStreamInfer);
-
-  LOG_INFO << "Register Status RPC";
-  (*grpc_server)->rpcStatus_ = inferenceService->RegisterRPC<StatusContext>(
-      &GRPCService::AsyncService::RequestStatus);
-
-  LOG_INFO << "Register Profile RPC";
-  (*grpc_server)->rpcProfile_ = inferenceService->RegisterRPC<ProfileContext>(
-      &GRPCService::AsyncService::RequestProfile);
-
-  LOG_INFO << "Register Health RPC";
-  (*grpc_server)->rpcHealth_ = inferenceService->RegisterRPC<HealthContext>(
-      &GRPCService::AsyncService::RequestHealth);
+      server, 1 /* infer threads */, 1 /* mgmt threads */);
+  (*grpc_servers).clear();
+
+  for (auto const& ep_map : port_map) {
+    std::string addr = "0.0.0.0:" + std::to_string(ep_map.first);
+    LOG_INFO << "Starting a GRPCService at " << addr;
+
+    (*grpc_servers).push_back(nullptr);
```

Comment:
grpc_servers->push_back(new GRPCServer(addr))  ?
Might need to use emplace_back since it is unique_ptr